cmd/clusterctl/client/cluster/upgrader.go

```diff
@@ -358,6 +363,36 @@ func (u *providerUpgrader) doUpgrade(ctx context.Context, upgradePlan *UpgradePl
 		}
 	}
 
+	// Block unsupported skip upgrades for Core, Kubeadm Bootstrap, Kubeadm ControlPlane.
+	// NOTE: in future we might consider extending the clusterctl contract to support enforcing of skip upgrade
+	// rules for out of tree providers.
+	minVersionSkew := semver.MustParse("1.10.0")
+	for _, upgradeItem := range upgradePlan.Providers {
+		if !(upgradeItem.Type == string(clusterctlv1.CoreProviderType) ||
+			(upgradeItem.Type == string(clusterctlv1.BootstrapProviderType) && upgradeItem.ProviderName == config.KubeadmBootstrapProviderName) ||
+			(upgradeItem.Type == string(clusterctlv1.ControlPlaneProviderType) && upgradeItem.ProviderName == config.KubeadmBootstrapProviderName)) {
+			continue
+		}
+
+		currentVersion, err := semver.ParseTolerant(upgradeItem.Version)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse current version for %s provider", upgradeItem.InstanceName())
+		}
+
+		if currentVersion.LT(minVersionSkew) {
+			continue
+		}
+
+		nextVersion, err := semver.ParseTolerant(upgradeItem.NextVersion)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse next version for %s provider", upgradeItem.InstanceName())
+		}
+
+		if nextVersion.Minor >= currentVersion.Minor+3 {
```

Comment:
```suggestion
		if nextVersion.Minor > currentVersion.Minor+3 {
```
Current code would block a n-3 => n upgrade which we want to support

Reply:
Please also adjust unit test coverage accordingly (i.e. it should cover that n-3 => n is allowed)